### PR TITLE
[Refactoring] Moving tests from JUnit 4 to JUnit 5 in mailbox-api module

### PR DESCRIPTION
**Pull request by Arsnael**
Except for `AbstractSubscriptionManagerTest` and `MailboxManagerStressTest`.
Those are trickier and require quite some work with the implementation classes depending on it.
Will try moving them in an other PR, an other time.

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[ERROR] /james-project/mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/JPAMailboxManagerTest.java:[58,5] method does not override or implement a method from a supertype
```

**Comment by Arsnael** [comment]
```
[ad6e26a60d7eecc777fdf9540f682c9e97d820b9] [ERROR]   EventDeadLettersIntegrationTest.groupIdOfFailedEventShouldBeStoredInDeadLetter:240 1 expectation failed.
[ad6e26a60d7eecc777fdf9540f682c9e97d820b9] JSON path . doesn't match.
[ad6e26a60d7eecc777fdf9540f682c9e97d820b9] Expected: iterable with items ["org.apache.james.webadmin.integration.EventDeadLettersIntegrationTest$RetryEventsListenerGroup"] in any order
[ad6e26a60d7eecc777fdf9540f682c9e97d820b9]   Actual: []
```
test this please

File: mailbox/api/src/test/java/org/apache/james/mailbox/ApplicableFlagBuilderTest.java [hunk 1]
```diff
@@ -22,25 +22,25 @@
...
 
 import com.google.common.collect.ImmutableList;
 
-public class ApplicableFlagBuilderTest {
+class ApplicableFlagBuilderTest {
 
-    @Rule
-    public final JUnitSoftAssertions softly = new JUnitSoftAssertions();
+    @RegisterExtension
+    public final JUnitJupiterSoftAssertions softly = new JUnitJupiterSoftAssertions();
```
Comment (blackheaven) [inline mailbox/api/src/test/java/org/apache/james/mailbox/ApplicableFlagBuilderTest.java]: It worth rewriting them using the "new" way to do soft assertions

Comment (chibenwa) [reply mailbox/api/src/test/java/org/apache/james/mailbox/ApplicableFlagBuilderTest.java]: That is ?

Comment (Arsnael) [reply mailbox/api/src/test/java/org/apache/james/mailbox/ApplicableFlagBuilderTest.java]: Like this you mean?
```
SoftAssertions.assertSoftly(softly -> {
// soft asserts...
});
```

Alright!

Comment (Arsnael) [reply mailbox/api/src/test/java/org/apache/james/mailbox/ApplicableFlagBuilderTest.java]: Remembered doing the same in an other similar refactoring.... Just pushed an extra commit to fix it as well, thanks for the feedback

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ void tearDown() throws Exception {
     }
 
     @Test
-    void creatingConcurrentlyMailboxesWithSameParentShouldNotFail() throws Exception {
+    public void creatingConcurrentlyMailboxesWithSameParentShouldNotFail() throws Exception {
```
Comment (rouazana) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java]: protected?

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java [hunk 3]
```diff
@@ -144,7 +144,7 @@ void tearDown() throws Exception {
     }
 
     @Test
-    public void creatingConcurrentlyMailboxesWithSameParentShouldNotFail() throws Exception {
+    protected void creatingConcurrentlyMailboxesWithSameParentShouldNotFail() throws Exception {
```
Comment (blackheaven) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java]: nothing at all?

Comment (Arsnael) [reply mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java]: Doesn't work with nothing at all, as the test is being disabled on `JPAMailboxManagerTest` inheriting `MailboxManagerTest`

**Comment by Arsnael** [comment]
```
[ERROR]   EventDeadLettersIntegrationTest.failedEventShouldNotBeInDeadLettersAfterSuccessfulRedelivery:332->retrieveFirstFailedInsertionId:195 » IndexOutOfBounds
```
test this please

**Comment by chibenwa** [comment]
:green_apple: before CI restart

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**